main.py

```diff
@@ -648,6 +644,16 @@ async def scrape_channel_media(
                 )
             )
 
+    # Save all files if not in cache
+    async def dl_file(attachment: Attachment, attachment_filepath: str) -> None:
```

Comment:
Have you tried running this on 40 songs? I wonder if running 40 downloads concurrently might lead to issues.

I looked into ways to do batches of tasks, though it's a little cumbersome.

Reply:
I tested with the 30 songs from busty's 4, though you're probably right that I should test with at least 60 or so before merging

Reply:
If we wanted to limit concurrent downloads I think the "correct" primitive to use would be a semaphore, which seems to not involve too much code to set up. See the first code sample [here](https://tutorialedge.net/python/concurrency/python-asyncio-semaphores-tutorial/).

Reply:
That looks like a sane way of doing it, yes. And even if it works on your machine, it might be good to limit to ~10 songs anyhow. Our VPS has lots of bandwidth, but isn't massively powerful.